### ISSUE TITLE
Crash: SecurityException in ImageUtils.getWPImageSpanThumbnailFromFilePath

### ISSUE THREAD

**Issue by hypest** [wordpress-mobile/WordPress-Android#6583]
```
Fatal Exception: java.lang.SecurityException: Permission Denial: opening provider com.samsung.android.provider.memo.MemoProvider from ProcessRecord{2c7eb65d 23502:org.wordpress.android/u0a125} (pid=23502, uid=10125) requires com.samsung.android.memo.READ or com.samsung.android.memo.WRITE
       at android.os.Parcel.readException(Parcel.java:1546)
       at android.os.Parcel.readException(Parcel.java:1499)
       at android.app.ActivityManagerProxy.getContentProvider(ActivityManagerNative.java:3515)
       at android.app.ActivityThread.acquireProvider(ActivityThread.java:5062)
       at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider(ContextImpl.java:2736)
       at android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1446)
       at android.content.ContentResolver.query(ContentResolver.java:466)
       at android.content.ContentResolver.query(ContentResolver.java:426)
       at org.wordpress.android.util.MediaUtils.getDataColumn(MediaUtils.java:514)
       at org.wordpress.android.util.MediaUtils.getPath(MediaUtils.java:428)
       at org.wordpress.android.util.MediaUtils.getRealPathFromURI(MediaUtils.java:400)
       at org.wordpress.android.util.ImageUtils.resizeImageAndWriteToStream(ImageUtils.java:415)
       at org.wordpress.android.util.ImageUtils.createThumbnailFromUri(ImageUtils.java:652)
       at org.wordpress.android.util.ImageUtils.getWPImageSpanThumbnailFromFilePath(ImageUtils.java:364)
       at org.wordpress.android.ui.posts.adapters.PostsListAdapter.showFeaturedImage(PostsListAdapter.java:318)
       at org.wordpress.android.ui.posts.adapters.PostsListAdapter.onBindViewHolder(PostsListAdapter.java:230)
       at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6356)
       at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6389)
       at android.support.v7.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5335)
       at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5598)
       at android.support.v7.widget.GapWorker.prefetchPositionWithDeadline(GapWorker.java:282)
       at android.support.v7.widget.GapWorker.flushTaskWithDeadline(GapWorker.java:336)
       at android.support.v7.widget.GapWorker.flushTasksWithDeadline(GapWorker.java:349)
       at android.support.v7.widget.GapWorker.prefetch(GapWorker.java:356)
       at android.support.v7.widget.GapWorker.run(GapWorker.java:387)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5910)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1405)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1200)```

**Comment by daniloercoli** [comment]
Do we know the magnitude of this crash? Would be cool to have a link the original Crashlitycs issue.

I'm wondering how a `content://...` picture ended up to be linked in the feature image field for a post. It seems to me that FI on Android can only be set by using a remote picture.
We've seen this security issue many times in the past, and it was due to download a media - that was not in internal media store - on a background thread.  
In any case, I didn't find a way to set a FI from a local picture. @nbradbury do you know if there is a way of using a local picture as FI?

**Comment by hypest** [comment]
Here's an internal link to the report @daniloercoli https://a8c.slack.com/archives/C02QANACA/p1503674445000332

**Comment by daniloercoli** [comment]
Thanks @hypest - I see some other places in the code that could lead to this error (even though we should had already fixed them).
My original question instead still stand. I can't find a way to add a FI from a remote picture.

**Comment by daniloercoli** [comment]
I've checked this issue once again today, and can't find a way to repro it. I can only set FI from WP media library, but all the 3rd party pictures I shared to WP were downloaded locally, by checking if the picture is already in media store ->`MediaUtils.isInMediaStore`. So now way for me to set a feature image with `content://evernote...` or `content://com.samsung` as source.

Actually it seems that `8.1` is not affected by this issue. Maybe is too early to speak, crossing fingers.

**Comment by hypest** [comment]
Still happening, at least up to the current latest version (v9.4) of the app.

**Comment by daniloercoli** [comment]
I forgot to close this issue when merged #7748.

As I reported in the other ticket the crash happens at `at org.wordpress.android.util.MediaUtils.getDataColumn(MediaUtils.java:504)` when the post content contained media in a bad state. Not quite sure how I ended up with a post with a raw picture in it, but I guess it's easy if you have .raw pictures stored on your device, in the Download folder in my case, and then try to attach it to a post you can make the app crashing once re-openig the post: `content://com.android.providers.downloads.documents/document/raw%3A%2Fstorage%2Femulated%2F0%2FDownload%2FIMG_3224.JPG`

In my case it was a .JPG file returned from the `document` content provider.
